### ISSUE TITLE
Usage of tslearn.utils.to_time_series_dataset gives error

### ISSUE THREAD

**Issue by 12sam12** [tslearn-team/tslearn#335]
**Describe the bug**
Unable to use to_time_series_dataset method. Following is the error thrown when the dataframe has any date column:
TypeError: float() argument must be a string or a number, not 'Timestamp'

**To Reproduce**
A short code sample to reproduce the code

**Expected behavior**
exceute properly

**Environment (please complete the following information):**
 - OS: Ubuntu
 - tslearn version latest

**Additional context**

**Comment by GillesVandewiele** [comment]
Tslearn expects timeseries with a fixed sample rate (i.e. the difference between consecutive timestamps should be the same). As such, including a timestamp has not much use.

You can, however, convert the timestamps to numbers. E.g. conversion to UTC timestamp.

**Comment by 12sam12** [comment]
Yes I agree that. But I thought the final result would have this date column to interpret the results easily if it was fine to pass them as well.

**Comment by 12sam12** [comment]
In some machine learning algorithms, if you have X and y kinds of data kind of a supervised learning setup, in those cases. Lets assume a dataset where an adjusted close price is a prediction feature. In that as Im just focussed on that feature, what will go into my X and y?
